src/context/abstractCast.ml

```diff
@@ -7,25 +7,36 @@ open Error
 
 let cast_stack = new_rec_stack()
 
-let make_static_call ctx c cf a pl args t p =
+let rec make_static_call ctx c cf a pl args t p =
 	if cf.cf_kind = Method MethMacro then begin
 		match args with
 			| [e] ->
 				let e,f = push_this ctx e in
 				ctx.with_type_stack <- (WithType.with_type t) :: ctx.with_type_stack;
 				let e = match ctx.g.do_macro ctx MExpr c.cl_path cf.cf_name [e] p with
-					| Some e -> type_expr ctx e WithType.value
+					| Some e -> type_expr ctx e (WithType.with_type t)
 					| None ->  type_expr ctx (EConst (Ident "null"),p) WithType.value
 				in
+				let e = cast_or_unify ctx t e p in
```

Comment:
Do we really want `cast_or_unify` here? That's technically a transitive implicit cast, right? I think just unifying here is better.

Reply:
Yes, we do want that. Because non-macro froms allow to return a value, which will be handled by another `@:from`
```haxe
abstract A(Int) {
	@:from static function fromBool(v:Bool):A {
		return '$v';
	}

	@:from static function fromString(v:String):A {
		return new A(v == 'false' ? 0 : 1);
	}

	function new(v:Int) this = v;
}
```